sympy/logic/boolalg.py

```diff
@@ -521,15 +521,15 @@ def _to_nnf(cls, *args, **kwargs):
     @classmethod
     def _to_anf(cls, *args, **kwargs):
         deep = kwargs.get('deep', True)
-        argset = set()
+        new_args = []
         for arg in args:
             if deep:
                 if not is_literal(arg) or isinstance(arg, Not):
                     arg = arg.to_anf(deep=deep)
-                argset.add(arg)
+                new_args.append(arg)
             else:
-                argset.add(arg)
-        return cls(*argset, remove_true=False)
+                new_args.append(arg)
```

Comment:
I don't think `else` is necessary because `new_args.append(arg)` is executed whether `deep` is True or False.

```python
if deep:
    ...
new_args.append(arg)
```